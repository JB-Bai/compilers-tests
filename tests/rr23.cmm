float C3pZP8yiTP, RCuVg;
int edeQbs4oL(){
	struct lRcRT{
	}
	b4k3bNci = AK6Pv && 969, GNvqLxYa;
	295;
}
int vEJmiHcDX, c2OAij;
struct YnvQWkFU1 siNHgovN(int wVsm8_){
	return 6609.acjs;
	Z4cGyn2;
}
int zEdf3(){
	46.67E+9;
}

