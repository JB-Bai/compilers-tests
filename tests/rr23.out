Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: float
      ExtDecList (1)
        VarDec (1)
          ID: C3pZP8yiTP
        COMMA
        ExtDecList (1)
          VarDec (1)
            ID: RCuVg
      SEMI
    ExtDefList (2)
      ExtDef (2)
        Specifier (2)
          TYPE: int
        FunDec (2)
          ID: edeQbs4oL
          LP
          RP
        CompSt (2)
          LC
          DefList (3)
            Def (3)
              Specifier (3)
                StructSpecifier (3)
                  STRUCT
                  OptTag (3)
                    ID: lRcRT
                  LC
                  RC
              DecList (5)
                Dec (5)
                  VarDec (5)
                    ID: b4k3bNci
                  ASSIGNOP
                  Exp (5)
                    Exp (5)
                      ID: AK6Pv
                    AND
                    Exp (5)
                      INT: 969
                COMMA
                DecList (5)
                  Dec (5)
                    VarDec (5)
                      ID: GNvqLxYa
              SEMI
          StmtList (6)
            Stmt (6)
              Exp (6)
                INT: 295
              SEMI
          RC
      ExtDefList (8)
        ExtDef (8)
          Specifier (8)
            TYPE: int
          ExtDecList (8)
            VarDec (8)
              ID: vEJmiHcDX
            COMMA
            ExtDecList (8)
              VarDec (8)
                ID: c2OAij
          SEMI
        ExtDefList (9)
          ExtDef (9)
            Specifier (9)
              StructSpecifier (9)
                STRUCT
                Tag (9)
                  ID: YnvQWkFU1
            FunDec (9)
              ID: siNHgovN
              LP
              VarList (9)
                ParamDec (9)
                  Specifier (9)
                    TYPE: int
                  VarDec (9)
                    ID: wVsm8_
              RP
            CompSt (9)
              LC
              StmtList (10)
                Stmt (10)
                  RETURN
                  Exp (10)
                    Exp (10)
                      INT: 6609
                    DOT
                    ID: acjs
                  SEMI
                StmtList (11)
                  Stmt (11)
                    Exp (11)
                      ID: Z4cGyn2
                    SEMI
              RC
          ExtDefList (13)
            ExtDef (13)
              Specifier (13)
                TYPE: int
              FunDec (13)
                ID: zEdf3
                LP
                RP
              CompSt (13)
                LC
                StmtList (14)
                  Stmt (14)
                    Exp (14)
                      FLOAT: 46670000128.000000
                    SEMI
                RC
